src/fides/api/task/create_request_tasks.py

```diff
@@ -191,9 +193,21 @@ def base_task_data(
             # Serialize with duck typing so we get the nested sub fields as well
             dataset_graph.nodes[node].collection.model_dump_json(serialize_as_any=True)
         )
+
         # Saves traversal details based on data dependencies like incoming edges
         # and input keys, also useful for building the Execution Node
-        traversal_details = traversal_nodes[node].format_traversal_details_for_save()
+        if node in traversal_nodes:
+            traversal_details = traversal_nodes[
+                node
+            ].format_traversal_details_for_save()
+        else:
+            # If node is not in traversal_nodes, then it is a node added for
+            # custom request field processing. We manually build the traversal details,
+            # with no incoming or outgoing edges and no input keys.
+            graph_node = dataset_graph.nodes[node]
+            traversal_details = TraversalDetails.create_empty_traversal(
+                graph_node.dataset.connection_key
+            ).model_dump(mode="json")
```

Comment:
this was needed so the unreachable nodes wouldn't break the request tasks... not the cleanest solution but I couldn't find a better one so suggestions are welcome

Reply:
The big thing here would be making these nodes reachable in the first place unfortunately